aten/src/ATen/core/op_registration/op_registration.cpp

```diff
@@ -153,27 +153,50 @@ namespace {
   }
 }
 
-Module&& Module::def(const char* schema) && {
-  register_.op(c10::RegisterOperators::options()
-    .schema(addNamespace(ns_, schema))
-    .aliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA));
+// TODO: add overloads that take FunctionSchema directly
+Module&& Module::def(const char* schema_str) && {
+  auto schema = torch::jit::parseSchema(addNamespace(ns_, schema_str));
+  schema.setAliasAnalysis(c10::AliasAnalysisKind::FROM_SCHEMA); // TODO: clean this up
+  registrars_.emplace_back(Dispatcher::singleton().registerDef(std::move(schema)));
```

Comment:
nit: can probably be `push_back`.

Reply:
Isn't it a wash here? In both cases you're going to the call the move constructor to setup the entry in `registrars_`

Reply:
yes it probably is a wash but push_back is the simpler mechanism. It's a nit though, feel free to keep as is.